aten/src/ATen/native/Loss.cpp

```diff
@@ -199,6 +199,39 @@ Tensor& binary_cross_entropy_backward_out_cpu(Tensor& grad_input, const Tensor&
     return grad_input;
 }
 
+Tensor binary_cross_entropy_target_backward(const Tensor &grad, const Tensor& input, const Tensor& weight, int64_t reduction) {
+    Tensor grad_input = at::empty_like(input);
+
+    // These squeezes are copied from binary_cross_entropy_backward_out_cpu backward
+    auto iter = TensorIteratorConfig()
+      .add_output(grad_input)
+      .add_input(at::squeeze(grad))
+      .add_input(at::squeeze(input))
+      .build();
+
+    AT_DISPATCH_FLOATING_TYPES(input.scalar_type(), "binary_cross_entropy_target_backward", [&] {
+        at::native::cpu_kernel(
```

Comment:
is this actually CPU-only?
Would it be simpler to have a generic implementation in `FunctionsManual.cpp` that will work for all device?

Reply:
it does look that way -- a little scary that the tests didn't complain.